src/ResourceStore.php

```diff
@@ -0,0 +1,237 @@
+<?php
+
+namespace silverorange;
+
+use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\Client as HttpClient;
+
+class ResourceStore
+{
+    // {{{ protected properties
+
+    protected $json_api_base;
+    protected $json_api_headers;
+
+    protected $http_client;
+
+    protected $class_by_type = array();
+
+    protected $resources = array();
+
+    // }}}
+    // {{{ public function __construct()
+
+    public function __construct($json_api_base, $json_api_headers = [])
```

Comment:
The reason I went with this approach was so that the HTTP client could be recreated when this object is unserialized.